src/main/java/io/pinecone/configs/PineconeConfig.java

```diff
@@ -0,0 +1,63 @@
+package io.pinecone.configs;
+
+import io.grpc.ManagedChannel;
+import io.pinecone.exceptions.PineconeValidationException;
+
+public class PineconeConfig {
+
+    private String apiKey;
+    private String host;
+    private String usageContext;
```

Comment:
We decided to add this for detecting user-agent string coming from spark connector. Initially it was added this in the PineconeClientConfig class.
https://github.com/pinecone-io/pinecone-java-client/pull/43

Reply:
What do you think of naming this `integrationId`? That's what I went with when I implemented the same thing on the TypeScript side, although it hasn't landed yet: https://github.com/pinecone-io/pinecone-ts-client/pull/197

I think `usageContext` ends up feeling more broad, or certain users may see `context` and make assumptions about usage. We should probably also align the value in user-agents across clients. You'd have to update the `getUserAgent()` call as well, but everything else could stay the same.

Reply:
I agree, given usage now means something, its not a good variable name. But doesn't integrationId refer to some sort of Id?

Reply:
I feel like `integration` should be represented in the name, since that's the intent of the field. `userContext` could still be misconstrued by users, and from my understanding this was meant to support passing specific integrations (`spark-connector`, `langchain`, etc) for us to evaluate usage.

Reply:
I think I also prefer `integrationId`. This is really aimed at use cases where partners like `langchain` or `llamaindex` want to build something on top of our stuff. Or high-level tools we choose to build ourselves on top of SDKs (e.g. canopy). `userContext` makes it seem like it's something for the typical end-user, but it's really not.

Reply:
Okay I'll make the change in next PR. I'll start an internal conversation regarding renaming clients as well.